cirq-core/cirq/transformers/gauge_compiling/gauge_compiling.py

```diff
@@ -201,6 +209,102 @@ def __call__(
+        symbols_by_loc: Dict[Tuple[str, ops.Qid, int], Dict[str, sympy.Symbol]] = {}
+
+        def single_qubit_next_symbol() -> Dict[str, sympy.Symbol]:
+            sid = next(symbol_count)
+            return _parameterize(1, sid)
+
+        # Build parameterized circuit.
+        for moment_id, moment in enumerate(circuit):
+            center_moment: List[ops.Operation] = []
+            left_moment: List[ops.Operation] = []
+            right_moment: List[ops.Operation] = []
+            for op in moment:
+                if isinstance(op, ops.TaggedOperation) and set(op.tags).intersection(
+                    context.tags_to_ignore
+                ):
+                    center_moment.append(op)
+                    continue
+                if op.gate is not None and op in self.target:
+                    # Build symbols for the gauge, for a 2-qubit gauge, symbols will be built for
+                    # pre/post q0/q1 and the new 2-qubit gate if the 2-qubit gate is updated in
+                    # the gauge compiling.
+                    center_moment.append(op)
+                    for prefix, q in itertools.product(["pre", "post"], op.qubits):
+                        xza_by_symbols = single_qubit_next_symbol()  # xza in phased xz gate.
+                        loc = (prefix, q, moment_id)
+                        symbols_by_loc[loc] = xza_by_symbols
+                        new_op = ops.PhasedXZGate(**xza_by_symbols).on(q)
+                        if prefix == "pre":
+                            left_moment.append(new_op)
+                        elif prefix == "post":
+                            right_moment.append(new_op)
+                else:
+                    center_moment.append(op)
+            new_moments.extend(
+                [moment for moment in [left_moment, center_moment, right_moment] if moment]
+            )
+
+        # Assign values for parameters via randomly chosen GaugeSelector.
+        for sample_id in range(N):
+            for moment_id, moment in enumerate(circuit):
+                for op in moment:
+                    if isinstance(op, ops.TaggedOperation) and set(op.tags).intersection(
+                        context.tags_to_ignore
+                    ):
+                        continue
+                    if op.gate is not None and len(op.qubits) == 2 and op in self.target:
+                        gauge = self.gauge_selector(rng).sample(op.gate, rng)
+                        if not gauge.support_randomized_compiling:
+                            raise NotImplementedError(f"{gauge.two_qubit_gate}")
+                        # Get the params of pre/post q0/q1 gates.
+                        for pre_or_post, idx in itertools.product(["pre", "post"], [0, 1]):
+                            symbols = symbols_by_loc[(pre_or_post, op.qubits[idx], moment_id)]
+                            gates = getattr(gauge, f"{pre_or_post}_q{idx}")
```

Comment:
abstract this code into a function `gate_sequence_to_sweep`. A gate sequence can be replaced by a single PhasedXZ gate

Reply:
Done.